tests/test_plugin.py

```diff
@@ -314,6 +315,133 @@ def test_add_and_cancel_child_plugin(self):
+        """
+        simple_page = create_page('test page', 'page.html', u'en')
+        simple_placeholder = simple_page.get_placeholders('en').get(slot='content')
+
+        # 2 Successes, position 1 and 2
+        add_plugin(
+            simple_placeholder,
+            "TextPlugin",
+            "en",
+            body="I'm the first",
+        )
+        add_plugin(
+            simple_placeholder,
+            "TextPlugin",
+            "en",
+            body="I'm the first",
+        )
+
+        # 1 Failure, position 3
+        endpoint = self.get_add_plugin_uri(simple_placeholder, 'TextPlugin')
+        with self.login_user_context(self.get_superuser()):
+            fail_1_response = self.client.get(endpoint)
+
+        # 1 Success, position 4
+        add_plugin(
+            simple_placeholder,
+            "TextPlugin",
+            "en",
+            body="I'm the first",
+        )
+
+        # We simulate the fact that the FE can only count actual plugins and the 3rd is a ghost,
+        # The value calculated would be 3 actual plugins so the next position is 4, we know that 4 is actually
+        # populated in this test which is what we need
+        endpoint.replace("plugin_position=3", "plugin_position=4")
+
+        self.assertEqual(CMSPlugin.objects.all().count(), 4)
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(endpoint)
+
+        self.assertEqual(response.status_code, 200)
+        # 1 should have been deleted and a new one added!
+        self.assertEqual(CMSPlugin.objects.all().count(), 4)
+
+    def test_add_and_cancel_plugin_when_plugin_position_is_taken(self):
+        """
+        A text plugin that already exists in a position returns an error that the plugin
+        position is populated and conflicts i.e. a plugin cannot be created
+        """
+        simple_page = create_page('test page', 'page.html', u'en')
+        simple_placeholder = simple_page.get_placeholders('en').get(slot='content')
+        # Add a plugin
+        endpoint = self.get_add_plugin_uri(simple_placeholder, 'TextPlugin')
+        original_plugin = add_plugin(
+            simple_placeholder,
+            "TextPlugin",
+            "en",
+            body="I'm the first",
+        )
+
+        # Ensure the positions of the new plugin and the endpoint to add another are the same
+        self.assertEqual(original_plugin.position, 1)
+        self.assertTrue("plugin_position=1" in endpoint)
+
+        # Try and add the same plugin again
+        # Adding a plugin in the same location is not allowed
+        with self.login_user_context(self.get_superuser()):
+
+            with self.assertRaises(Exception) as error:
+                with transaction.atomic():
+                    readd_response = self.client.get(endpoint)
+
+        # FIXME: Breaks because the code catches and integrity and then allows one to be thrown again!
```

Comment:
If this issue still exists can we fix it before merging?

Reply:
Yup, it won't be going over until this test is fixed, it's blocking the pr.